src/lib/proxy-server.ts

```diff
@@ -35,93 +49,167 @@ async function getSiteByHost( domain: string ): Promise< SiteDetails | null > {
+	// On Windows, node doesn't throw an error if port is busy, so we use the net module to explicitly check
+	// if it's possible to establish a TCP connection to that port (meaning it's busy)
+	if ( process.platform === 'win32' ) {
+		await new Promise< void >( ( resolve, reject ) => {
+			const tester = createConnection( { port }, () => {
+				// If we can connect, port is in use
+				tester.end();
+				reject( new Error( 'EADDRINUSE' ) );
+			} );
+
+			tester.setTimeout( 1000, () => {
+				tester.destroy();
+				reject( new Error( 'EADDRINUSE' ) );
+			} );
+
+			tester.on( 'error', ( err ) => {
+				if ( isErrnoException( err ) && err.code === 'ECONNREFUSED' ) {
+					// Port is available
+					resolve();
+				} else {
+					reject( err );
+				}
 			} );
 		} );
+	}
 
-		// On Windows, node doesn't throw an error if port 80 is busy, so we use the net module to explicitly check
-		// if it's possible to establish a TCP connection to that port (meaning it's busy)
-		if ( process.platform === 'win32' ) {
+	return true;
+}
+
+/**
+ * Attempts to start the proxy servers on ports 80 and 443
+ * This requires admin/root privileges
+ */
+export async function startProxyServer(): Promise< boolean > {
+	try {
+		// Start HTTP server if not already running
+		if ( ! isHttpProxyRunning ) {
+			await checkPortInWindows( 80 );
+			httpProxyServer = http.createServer( ( req, res ) => handleProxyRequest( req, res, false ) );
 			await new Promise< void >( ( resolve, reject ) => {
-				const tester = createConnection( { port: 80 }, () => {
-					// If we can connect, port is in use
-					tester.end();
-					reject( new Error( 'EADDRINUSE' ) );
-				} );
-
-				tester.setTimeout( 1000, () => {
-					tester.destroy();
-					reject( new Error( 'EADDRINUSE' ) );
-				} );
-
-				tester.on( 'error', ( err ) => {
-					if ( isErrnoException( err ) && err.code === 'ECONNREFUSED' ) {
-						// Port is available
+				httpProxyServer!
+					.listen( 80, () => {
+						console.log( `HTTP Proxy server started on port 80` );
+						isHttpProxyRunning = true;
 						resolve();
-					} else {
+					} )
+					.on( 'error', ( err ) => {
+						console.error( `Error starting HTTP proxy server on port 80:`, err );
 						reject( err );
-					}
-				} );
+					} );
 			} );
 		}
 
-		await new Promise< void >( ( resolve, reject ) => {
-			proxyServer!
-				.listen( 80, () => {
-					console.log( `Proxy server started on port 80` );
-					isProxyRunning = true;
-					resolve();
-				} )
-				.on( 'error', ( err ) => {
-					console.log( err );
-					console.error( `Error starting proxy server on port 80:`, err );
-					reject( err );
-				} );
-		} );
+		// Start HTTPS server if not already running
+		if ( ! isHttpsProxyRunning ) {
+			await checkPortInWindows( 443 );
+			const defaultOptions = {
+				SNICallback: async (
+					servername: string,
+					cb: ( err: Error | null, ctx?: SecureContext ) => void
+				) => {
```

Comment:
```suggestion
			const defaultOptions: https.ServerOptions = {
				SNICallback: async ( servername, cb ) => {
```
Better type definition